frontend/src/metabase/reference/databases/TableList.jsx

```diff
@@ -59,26 +60,24 @@ export const separateTablesBySchema = (
   tables,
   createSchemaSeparator,
   createListItem,
-) =>
-  Object.values(tables)
-    .sort((table1, table2) =>
-      table1.schema_name > table2.schema_name
-        ? 1
-        : table1.schema_name === table2.schema_name
-        ? 0
-        : -1,
-    )
-    .map((table, index, sortedTables) => {
-      if (!table || !table.id || !table.name) {
-        return;
-      }
-      // add schema header for first element and if schema is different from previous
-      const previousTableId = Object.keys(sortedTables)[index - 1];
-      return index === 0 ||
-        sortedTables[previousTableId].schema_name !== table.schema_name
-        ? [createSchemaSeparator(table), createListItem(table, index)]
-        : createListItem(table, index);
-    });
+) => {
+  const sortedTables = _.chain(tables)
+    .sortBy(t => t.name)
+    .sortBy(t => t.schema_name)
+    .value();
+
+  return sortedTables.map((table, index, sortedTables) => {
+    if (!table || !table.id || !table.name) {
+      return;
+    }
+    // add schema header for first element and if schema is different from previous
+    const previousTableId = Object.keys(sortedTables)[index - 1];
```

Comment:
I know that was implemented this way but this looks complex. If I were implementing this part I'd just group by schema and add a separator if it's not the first schema. Something like this:

```
const createSchemaSeparator = schemaName => (
  <li className={R.schemaSeparator}>{schemaName}</li>
);
```

```
const tablesBySchema = _.chain(tables)
    .sortBy(t => t.name)
    .sortBy(t => t.schema_name)
    .groupBy(t => t.schema_name)
    .value();

return Object.entries(tablesBySchema).map(([schemaName, tables], index) => 
  <Fragment key={schemaName}>
   {index > 0 && createSchemaSeparator(schemaName)}
    {createListItem(table)}
  </Fragment>
);

```

Reply:
Thank you for the tip @ranquild 

I didn't want to touch that part because it wasn't a part of the actual fix. If it was any other piece of the codebase, I'd open a separate PR for it, but there's a chance `/reference`s are soon going to be removed or rewritten so I'd just leave this for now.

Reply:
lgtm